linkml/generators/rustgen/template.py

```diff
@@ -9,6 +10,78 @@
 from linkml.generators.common.template import TemplateModel, _render
 
 
+class ContainerType(Enum):
+    LIST = "list"
+    MAPPING = "mapping"
+
+class RustRange(BaseModel):
+    optional: bool = False
+    containerType: Optional[ContainerType] = None
+    has_default: bool = False
+    is_class_range: bool = False
+    is_reference: bool = False
+    box_needed: bool = False
+    has_class_subtypes: bool = False
+    child_ranges: Optional[List["RustRange"]] = None
+    type_: str
+    
+    def type_for_field(self):
+        tp = self.type_
+        if self.has_class_subtypes:
+            if not self.is_reference:
+                tp = f"{tp}OrSubtype"
+        if self.box_needed:
+            tp = f"Box<{tp}>"
+        if self.containerType == ContainerType.LIST:
+            tp = f"Vec<{tp}>"
+        elif self.containerType == ContainerType.MAPPING:
+            tp = f"HashMap<String, {tp}>"
+        if self.optional and self.containerType is None:
+            tp = f"Option<{tp}>"
+        return tp
+    
+    def type_for_trait(self, crateref: Optional[str], setter: bool=False):
```

Comment:
docstrings would be nice on these to help me and future people understand what they do, when they should be used, what the different considerations are that affect the result, etc. :)